### PR TITLE
Create sourcecode-deployment-guide.md

### PR DESCRIPTION
**Pull request by evanliuchina**
Tutorial:
Build NewChain source code and Setup a NewChain read-only node
从编译newchain源码到运行一个只读节点
read it: https://github.com/newtonproject/newchain-deploy/blob/1c6bcf489c3ea2434154156efdf84ee28e483b7a/sourcecode-deployment-guide.md

### CODE REVIEW

File: sourcecode-deployment-guide.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+$ cd newchain
+$ make
+```
+
+Done. You now have build/bin/geth.
+
+## Install geth to /usr/local/bin
+
+```
+$ cp ~/git/newchain/build/bin/geth /usr/local/bin/
```
Comment (VieYang) [inline sourcecode-deployment-guide.md:44]: use `sudo`

File: sourcecode-deployment-guide.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+```
+
+## Prepare the data folder
+
+Mount the data partition to, for example, /data/
+
+Create a data folder:
+
+```
+$ mkdir /data/newchain
```
Comment (VieYang) [inline sourcecode-deployment-guide.md:54]: `/data` is not the default directory，

```bash
sudo mkdir -p /data/newchain && sudo chown -R `whoami` /data/newchain
```

File: sourcecode-deployment-guide.md [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+StaticNodes = []
+TrustedNodes = []
+ListenAddr = ":38311"
+EnableMsgEvents = false
+
+[Dashboard]
+Host = "localhost"
+Port = 8080
+Refresh = 5000000000
+```
```
Comment (VieYang) [inline sourcecode-deployment-guide.md:178]: And then replace `evan` to your username,

```bash
sed -i "s/evan/`whoami`/g" ~/.newchain/mainnet/config.toml
```

File: sourcecode-deployment-guide.md (hunk 1 above)
Comment (evanliuchina) [reply sourcecode-deployment-guide.md]: sudo is added.

File: sourcecode-deployment-guide.md (hunk 2 above)
Comment (evanliuchina) [reply sourcecode-deployment-guide.md:52]: Mount the data partition to, for example, /data/, then create a sub-folder within it:
$ mkdir /data/newchain
Be careful about the access permissions of the data folder.

File: sourcecode-deployment-guide.md (hunk 3 above)
Comment (evanliuchina) [reply sourcecode-deployment-guide.md:176]: Use {USERNAME} for users to do it by themselves:
Create a config file ~/.newchain/mainnet/config.toml (replace {USERNAME} below with your login name):

**Comment by evanliuchina** [comment]
Submitted a new change: https://github.com/newtonproject/newchain-deploy/blob/1c6bcf489c3ea2434154156efdf84ee28e483b7a/sourcecode-deployment-guide.md

**Pull request merged**